src/commands/periscope/periscope.ts

```diff
@@ -0,0 +1,177 @@
+    await createPeriscopeWebView(cloudTarget.cloudResource.name, undefined, undefined, false);
+    return undefined;
+  }
+}
+
+async function prepareAKSPeriscopeDeploymetFile(
+  clusterStorageInfo: PeriscopeStorage
+): Promise<string | undefined> {
+  const tempFile = tmp.fileSync({ prefix: "aks-periscope-", postfix: `.yaml` });
+  const periscopeDeploymentFile = fs.createWriteStream(tempFile.name);
+
+  writeTempAKSDeploymentFile(periscopeDeploymentFile);
+
+  periscopeDeploymentFile.on('finish', async () => {
+    // Replace the storage account name and keys in persicope deplyment file.
+    replaceDeploymentAccountNameAndSas(clusterStorageInfo, tempFile?.name);
+  });
+
+  return tempFile.name;
+}
+
+export async function runAssociatedAKSPeriscopeCommand(
+  clusterStorageInfo: PeriscopeStorage,
+  aksPeriscopeFile: string,
+  clusterKubeConfig: string | undefined,
+  clusterName: string
+) {
+  const kubectl = await k8s.extension.kubectl.v1;
+
+  if (kubectl.available) {
+    // Clean up running instance.
+    await tmpfile.withOptionalTempFile<any>(
+      clusterKubeConfig, "YAML",
+      (f) => kubectl.api.invokeCommand(`delete ns aks-periscope --kubeconfig="${f}"`));
+
+    // Deploy the aks-periscope.
+    const runCommandResult = await tmpfile.withOptionalTempFile<any>(
+      clusterKubeConfig, "YAML",
+      (f) => kubectl.api.invokeCommand(`apply -f ${aksPeriscopeFile} --kubeconfig="${f}" && kubectl cluster-info --kubeconfig="${f}"`));
+
+    await createPeriscopeWebView(clusterName, runCommandResult, clusterStorageInfo);
+  }
+}
+
+async function createPeriscopeWebView(
+  clusterName: string,
+  outputResult: any,
+  periscopeStorageInfo: PeriscopeStorage | undefined,
+  hasDiagnosticSettings = true
+) {
+  const panel = vscode.window.createWebviewPanel(
+    `AKS Periscope`,
+    `AKS periscope: ${clusterName}`,
+    vscode.ViewColumn.Active,
+    {
+      enableScripts: true,
+      enableCommandUris: true
+    }
+  );
+
+  const extensionPath = getExtensionPath();
+
+  if (extensionPath && !hasDiagnosticSettings) {
+    // In case of no diagnostic setting we serve user with helpful content in webview and
+    // a link as to how to attach the storage account to cluster's diagnostic settings.
+    panel.webview.html = getWebviewContent(clusterName, extensionPath, outputResult, undefined, [], hasDiagnosticSettings);
+  }
+
+  if (extensionPath && periscopeStorageInfo) {
+    // For the case of successful run of the tool we render webview with the output information.
+    panel.webview.html = getWebviewContent(clusterName, extensionPath, outputResult, periscopeStorageInfo, []);
+
+    let downloadableAndShareableNodeLogsList: PeriscopeHTMLInterface[] | undefined;
+
+    panel.webview.onDidReceiveMessage(
+      async (message) => {
+        await longRunning(`Generating storage downloadable link.`,
+          async () => {
+            if (message.command === "generateDownloadLink") {
```

Comment:
💡 Cool,  I might get some thoughts shred's with you  tomorrow or early next week (if you have time) week once you have had chance to look at the current state incorporating all the changes, and see if this is in need now.

I kind of agree as well, but other chain of that is that since it is just used at this one place.I guess on the other hand it will make more sense if we can put some extension level constant.